docs/src/pattern-matching.md

```diff
@@ -97,18 +97,18 @@ which will match e.g. `struct Foo ...` but not `struct Foo{V} ...`
 for example:
 
 ```julia
-@capture(ex, f_(args__) = body_ | function f_(args__) body_ end)
+@capture(ex, (f_(args__) = body_) | (function f_(args__) body_ end))
 ```
 
 will match both kinds of function syntax (though it's easier to use
 `shortdef` to normalise definitions). You can also do this within
 expressions, e.g.
 
 ```julia
-@capture(ex, (f_{T_}|f_)(args__) = body_)
+@capture(ex, (f_(args__) where {T_}) | (f_(args__)) = body_)
```

Comment:
It's nice to have the fix here, but this example no longer shows what it was intended to (an `|` pattern inside another pattern). Would be nice to come up with an example where we can show that.

Reply:
umm I think
```jl
@capture(ex, (f_(args__) where {T_}) | (f_(args__)) = body_)
```
shows an union pattern in an expression (i.e. LHS expression in a shortform function def) ?

Reply:
Or, how about this ?
```jl
@capture((Base.isconst|isconst)(args__))
```
to capture both calls (but kind of an nonsense example)

Reply:
With `where` clause you will need `0:1` choice. That's will be very hard without epsilon transition to illustrate it via an `or` pattern. 

My advise keep the `func{params}(args)` which is still in use for constructor
(even considering https://github.com/JuliaLang/julia/issues/11310) to illustrate nested `or`.

Develop others forms notably with `where` as separated entries of a
```julia
@capture ex begin
pat1 => ...
pat2 => ...
end
```

Reply:
@MikeInnes

I still haven't came up with the good "an `|` pattern inside another pattern" example...
Could you give me any advice in order to finish up this PR, please ?

Reply:
Let's just stick to this, it's a little more complex but that's ok.